File: src/test/java/io/github/saneea/dvh/textfunction/String2CppArrayTest.java
package io.github.saneea.dvh.textfunction;

import io.github.saneea.dvh.FeatureContext;
import io.github.saneea.dvh.feature.text.ToCharArray;
import org.junit.Test;

import java.io.StringReader;
import java.io.StringWriter;

import static org.junit.Assert.assertEquals;

public class String2CppArrayTest {

    @Test
    public void testSimple() {
        test("abc", "'a', 'b', 'c'");
    }

    @Test
    public void testEscapingQuote() {
        test("a'bc", "'a', '\\'', 'b', 'c'");
    }

    @Test
    public void testEscapingBackslash() {
        test("a\\bc", "'a', '\\\\', 'b', 'c'");
    }

    private void test(String input, String expected) {
        ToCharArray toCharArray = new ToCharArray();
        toCharArray.setIn(new StringReader(input));
        StringWriter out = new StringWriter();
        toCharArray.setOut(out);

        FeatureContext featureContext = new FeatureContext(null, "any", new String[0]);
        toCharArray.run(featureContext);

        String actual = out.toString();
        assertEquals(expected, actual);
    }

}

File: src/test/java/io/github/saneea/dvh/textfunction/String2CppArrayTest.kt
package io.github.saneea.dvh.textfunction

import io.github.saneea.dvh.FeatureContext
import io.github.saneea.dvh.feature.text.ToCharArray
import org.junit.Assert
import org.junit.Test
import java.io.StringReader
import java.io.StringWriter

class String2CppArrayTest {

    @Test
    fun testSimple() = test("abc", "'a', 'b', 'c'")

    @Test
    fun testEscapingQuote() = test("a'bc", "'a', '\\'', 'b', 'c'")

    @Test
    fun testEscapingBackslash() = test("a\\bc", "'a', '\\\\', 'b', 'c'")

    private fun test(input: String, expected: String) {
        val toCharArray = ToCharArray()
        toCharArray.setInTextReader(StringReader(input))
        val out = StringWriter()
        toCharArray.outTextWriter = out

        val featureContext = FeatureContext(null, "any", emptyArray())
        toCharArray.run(featureContext)
        val actual = out.toString()
        Assert.assertEquals(expected, actual)
    }
}